ramips: add support for WIZnet WizFi630S board

7tobias opened the pull request:
The WIZnet WizFi63S board is in the miniPCIe form factor.

SoC: Mediatek MT7688AN
RAM: 128MB
Flash: 32Mb
WiFi: 2.4GHz
Ethernet: 3x 100Mbit
USB: 1 (USB 2.0)
serial ports: 2 (1x full, 1xlite)

Signed-off-by: Tobias Welz <tw@wiznet.eu>

nickberry17 commented:
Have you tested this on the latest master and found it to be working?  You might need to refactor the partitions into a 
`		partitions {
			compatible = "fixed-partitions";
			#address-cells = <1>;
			#size-cells = <1>;` 

structure (see: https://github.com/openwrt/openwrt/pull/1732/files) and add `compatible = "denx,uimage";` to your firmware partition definition.

ynezz commented:
@7tobias You're missing flashing instructions in your commit message, for details please read https://openwrt.org/submitting-patches

> If you add support for new hardware: Include in your commit message a short description of the hardware and how to install OpenWrt on it. Have a look at the [recent additions](https://git.openwrt.org/?p=openwrt/openwrt.git&a=search&h=HEAD&st=commit&s=add+support+for) for some examples.

ynezz commented on target/linux/ramips/base-files/lib/ramips.sh:
```diff
@@ -577,6 +577,9 @@ ramips_board_detect() {
 	*"WizFi630A")
 		name="wizfi630a"
 		;;
+	*"WizFi630S")
```
You can drop this, the board name is now taken from DTS.

ynezz commented on target/linux/ramips/base-files/etc/board.d/02_network:
```diff
@@ -416,6 +416,10 @@ ramips_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0:lan" "1:lan" "2:wan" "6@eth0"
 		;;
+	wizfi630s)
```
Change it to `wiznet,wizfi630s` as specified in DTS.

7tobias replied:
if i change it this way - the switch disappears. Idk how to resolve this, instead of keeping this line as it is.

7tobias replied:
Found it - switch is fine again - forgot to remove the line in ramips.sh

ynezz commented on target/linux/ramips/dts/WIZFI630S.dts:
```diff
@@ -0,0 +1,196 @@
+/dts-v1/;
```
@7tobias Please can you consider `SPDX-License-Identifier: GPL-2.0-or-later OR MIT` as in [DTS Checklist](https://openwrt.org/submitting-patches#dts_checklist) ?

> Having MIT as alternative is sometimes preferred by upstream maintainers
and allows sharing that simple code with other projects. We don't really
want multiple DTS versions for the same device.

ynezz commented on target/linux/ramips/dts/WIZFI630S.dts:
```diff
@@ -0,0 +1,196 @@
+                led-upgrade = &led_run;
+        };
+
+	gpio-leds {
```
@7tobias You've some white space issue here. While at it could you please change this to generic name `leds` as suggested by [DTS Checklist](https://openwrt.org/submitting-patches#dts_checklist) ?
>  5. The name of a node should reflect the function of the device and not its model. Examples for generic node names can be found in Section 2.2.2 Generic Names Recommendation

ynezz commented on target/linux/ramips/dts/WIZFI630S.dts:
```diff
@@ -0,0 +1,196 @@
+		};
+	};
+
+	gpio-keys-polled {
```
@7tobias  Could you please change this to generic name `keys` as suggested by [DTS Checklist](https://openwrt.org/submitting-patches#dts_checklist)
>  5. The name of a node should reflect the function of the device and not its model. Examples for generic node names can be found in Section 2.2.2 Generic Names Recommendation

mrkiko commented:
Thank you for your work!
Does this device offer any recovery measure? May you elaborate it?
Thank you very much again.

ynezz commented on target/linux/ramips/dts/WIZFI630S.dts:
```diff
@@ -0,0 +1,196 @@
+		};
+	};
+
+	spidev@1 {
```
@7tobias What is this for? Can't find it in your hardware description.

7tobias commented:
> Have you tested this on the latest master and found it to be working? You might need to refactor the partitions into a
> `	partitions { compatible = "fixed-partitions"; #address-cells = <1>; #size-cells = <1>;`
> 
> structure (see: https://github.com/openwrt/openwrt/pull/1732/files) and add `compatible = "denx,uimage";` to your firmware partition definition.

I implemented all the comments and tested with latest master - it's working fine.

ynezz commented:
@7tobias Please can you squash/merge those two commits into the one?

```
git reset --soft HEAD~2
git commit -a -c ORIG_HEAD
```

or

```
git rebase -i HEAD~2
# now select the last commit and change `pickup` to `squash`
```

7tobias committed:
ramips: add support for WIZnet WizFi630S board

The WIZnet WizFi630S board is in the miniPCIe form factor.

SoC: Mediatek MT7688AN
RAM: 128MB
Flash: 32Mb
WiFi: 2.4GHz
Ethernet: 3x 100Mbit
USB: 1 (USB 2.0)
serial ports: 2 (1x full, 1xlite)

Flash and recovery instructions:
Use the factory installed u-boot boot loader. It is available on UART2 (115200,8,n,1).  Then get the sysupgrade image from a tftp server.

Signed-off-by: Tobias Welz <tw@wiznet.eu>

7tobias commented:
> @7tobias Please can you squash/merge those two commits into the one?
> 
> ```
> git reset --soft HEAD~2
> git commit -a -c ORIG_HEAD
> ```
> or
> 
> ```
> git rebase -i HEAD~2
> # now select the last commit and change `pickup` to `squash`
> ```

I took the sqashing advice from  https://openwrt.org/submitting-patches with a forced push - but your lines look more convenient. - maybe an alternative squashing methid for the wiki?

ynezz commented:
@7tobias  I've merged this with some small cosmetic changes into my [staging tree](https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=commit;h=a33a9fcf4a50c0ffd8442c8a88338fed7e838902) can you please check if it's ok?

7tobias commented:
> @7tobias I've merged this with some small cosmetic changes into my [staging tree](https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=commit;h=a33a9fcf4a50c0ffd8442c8a88338fed7e838902) can you please check if it's ok?

I took your staging tree, it compiled, i was able to flash it - module seems to be working fine with it.

ynezz commented:
Thanks!

adrianschmutzler commented:
Hi, I'm currently reviewing the MAC address assignment on ramips target.
For your device, you are currently using MAC address from <&factory 0x28> and then just calculate WAN address by adding +1 to that.
However, there might be another MAC address stored on flash for your device in <&factory 0x2e>.

Can you check that? If yes, I would also be interested in which of the two addresses should be "lan" and which "wan" then ...

7tobias commented:
> Hi, I'm currently reviewing the MAC address assignment on ramips target.
> For your device, you are currently using MAC address from <&factory 0x28> and then just calculate WAN address by adding +1 to that.
> However, there might be another MAC address stored on flash for your device in <&factory 0x2e>.
> 
> Can you check that? If yes, I would also be interested in which of the two addresses should be "lan" and which "wan" then ...

![image](https://user-images.githubusercontent.com/2125300/89318387-04539a80-d67f-11ea-8b26-7eda435c2030.png)
Yes you are right - there is one at 0x28, 0x2e and even one at 0x04 which is also on the label.  I think the label mac makes most sense for wan. I'll add that to my latest PR.

adrianschmutzler commented:
Thanks, but please don't redistribute the addresses based on your taste. Please assign the addresses as vendor firmware does it.

The address at 0x4 is the one taken for Wifi when reading caldata from 0x0, so that's implicitly used there.
